pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -172,77 +179,87 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 			r.status.OnCRNotFound()
 			return reconcile.Result{}, nil
 		}
-		reqLogger.Error(err, "Error querying for Application Layer")
-		r.status.SetDegraded("Error querying for Application Layer", err.Error())
+		status.SetDegraded(r.status, operatorv1.ResourceReadError, "Error querying for Application Layer", err, reqLogger)
 		return reconcile.Result{}, err
 	}
 	r.status.OnCRFound()
+	// SetMetaData in the TigeraStatus like observedGenerations.
+	if applicationLayer != nil {
```

Comment:
After OnCRFound we no longer need the nil check. These nil checks can be removed from all the controllers where OnCRFound is called.

Reply:
Done